src/main/java/com/edmunds/rest/databricks/restclient/DatabricksRestClientImpl.java

```diff
@@ -48,65 +49,51 @@
 
   private static Logger logger = Logger.getLogger(DatabricksRestClientImpl.class.getName());
 
-  private DatabricksRestClientImpl(String host, String apiVersion, int maxRetry,
-      long retryInterval) {
-    super(host, apiVersion, maxRetry, retryInterval);
-  }
+  private DatabricksServiceFactory.Builder builder;
 
   /**
-   * Constructs a rest client with user and password authentication.
+   * Constructs a rest client.
    */
-  public static DatabricksRestClientImpl createClientWithUserPassword(String username,
-      String password, String host,
-      String apiVersion, int maxRetry, long retryInterval) {
-    DatabricksRestClientImpl client = new DatabricksRestClientImpl(host, apiVersion, maxRetry,
-        retryInterval);
-    client.initClientWithUserPassword(username, password);
-    return client;
-  }
+  public DatabricksRestClientImpl(DatabricksServiceFactory.Builder builder) {
+    super(builder.getHost(), builder.getApiVersion(), builder.getMaxRetries(), builder.getRetryInterval());
+
+    this.builder = builder;
+    if (isNotEmpty(builder.getToken())
+            || (isNotEmpty(builder.getUsername()) && isNotEmpty(builder.getPassword()))) {
+      initClient();
```

Comment:
how about just pass builder to initClient?

Reply:
changed